src/locksManager.js

```diff
@@ -54,4 +54,20 @@ var LocksManager = GObject.registerClass({
         const variant = new GLib.Variant('a{sb}', {locked: false});
         this._proxy.SetSync(lock, variant);
     }
+
+    getTrapSequence(lock) {
+        return this._proxy.getDictValueSync(lock, 'trap_sequence');
+    }
+
+    setLockTried(lock, tried) {
+        try {
+            const [variant] = this._proxy.GetSync(lock);
+            const dict = new GLib.VariantDict(variant);
+            dict.insert_value('tried', new GLib.Variant('b', tried));
+            this._proxy.SetSync(lock, dict.end());
```

Comment:
Do we need to have this be sync? Can we just fire-and-forget an async operation instead?